commands/out.go

```diff
@@ -238,6 +250,158 @@ func put(req resource.OutRequest, img partial.WithRawManifest, tags []name.Tag,
+		Attachment:        "",
+		SkipConfirmation:  true,
+		TlogUpload:        false,
+		TSAServerURL:      "",
+		IssueCertificate:  false,
+		Fulcio: options.FulcioOptions{
+			URL: options.DefaultFulcioURL,
+		},
+		OIDC: options.OIDCOptions{
+			Issuer:      options.DefaultOIDCIssuerURL,
+			ClientID:    "sigstore",
+			RedirectURL: "",
+		},
+		Rekor: options.RekorOptions{
+			URL: options.DefaultRekorURL,
+		},
+		Registry: options.RegistryOptions{
+			AllowInsecure:      false,
+			KubernetesKeychain: false,
+			Keychain:           keychain,
+		},
+	}
+
+	ko := options.KeyOpts{
+		KeyRef:                         "env://COSIGN_KEY",
+		PassFunc:                       generate.GetPass,
+		Sk:                             false,
+		Slot:                           "",
+		FulcioURL:                      options.DefaultFulcioURL,
+		IDToken:                        "",
+		InsecureSkipFulcioVerify:       false,
+		RekorURL:                       options.DefaultRekorURL,
+		OIDCIssuer:                     options.DefaultOIDCIssuerURL,
+		OIDCClientID:                   "sigstore",
+		OIDCClientSecret:               "",
+		OIDCRedirectURL:                "",
+		OIDCDisableProviders:           false,
+		OIDCProvider:                   "",
+		SkipConfirmation:               true,
+		TSAServerURL:                   "",
+		IssueCertificateForExistingKey: false,
+	}
+
+	// because cosign is a CLI tool primiarly (until it has matured as a library)
+	// we have to set the COSIGN_KEY environment variable and tell cosign to use it.
+	// as it is less tricky than having to create files that contain the cosign.key
+	// which actually becomes less secure because anyone who gains access to the container
+	// can easily easily see the file. whereas the `os.Setenv` function does not permanently
+	// set environment variables for the parent process that runs it, only for child processes.
+	// however, it is still not perfect, we would rather pass the key directly to cosign but
+	// untill that functionality is offered this is the best option without making decisions on
+	// tooling (Vault, Azure KeyVault etc)
+	err = os.Setenv("COSIGN_KEY", req.Source.Cosign.Key)
+	if err != nil {
+		return fmt.Errorf("err %w", err)
+	}
+
+	// similiar to the COSIGN_KEY variable we set the password that was used to create the
+	// keypair (if there was one). there are less ways to set the password currently outside
+	// of an environment variable or user input via a terminal prompt, as the Cosign library
+	// evolves over time, we can reasonably expect this to change, but as we cannot rely on user
+	// input, we have to use the environment variable.
+	err = os.Setenv("COSIGN_PASSWORD", req.Source.Cosign.Password)
+	if err != nil {
+		return fmt.Errorf("err %w", err)
+	}
+
+	logrus.Infof("Signing image with Cosign: %s", imgDigestUrl)
+	err = sign.SignCmd(ro, ko, o, []string{imgDigestUrl})
+	if err != nil {
+		return fmt.Errorf("there was an error signing the image with Cosign %w", err)
```

Comment:
nit: add a colon to separate our error message and the upstream error message
```suggestion
		return fmt.Errorf("there was an error signing the image with Cosign: %w", err)
```

Reply:
:+1: